### ISSUE TITLE
[BUG] _vector_query parsing error preventing use of arguments such as distance_threshold

### ISSUE THREAD

**Issue by PeterThramkrongart** [typesense/typesense#1901]
## Description

There seems to be a bug in the argument parser of _vector_search when reranking a text query with vector search. If you want to apply a distance threshold to the vector search, it triggers a parsing error. The bug prevents the use of all arguments beside the embedding itself being used in _vector_query.

## Steps to reproduce

```
#python


import requests
import os
from dotenv import load_dotenv

# Load environment variables
load_dotenv()

TYPESENSE_API_KEY = os.getenv('TYPESENSE_API_KEY')
TYPESENSE_NODE = os.getenv('TYPESENSE_NODE')
TYPESENSE_INDEX = os.getenv('TYPESENSE_INDEX')

HEADERS = {
    "x-typesense-api-key": TYPESENSE_API_KEY
}

def test_vector_query():
    vector = [0.0]*5
    vector_str = ",".join(map(str, vector))

    url_0 = (
        f"https://{TYPESENSE_NODE}/collections/{TYPESENSE_INDEX}/documents/search"
        f"?query_by=name"
        f"&sort_by=_text_match:desc,_vector_query(Embedding:([{vector_str}]),distance_threshold:0.3):asc,rank_score:desc"
        f"&exclude_fields=Embedding"
        f"&q=*"
    )

    url_1 = (
        f"https://{TYPESENSE_NODE}/collections/{TYPESENSE_INDEX}/documents/search"
        f"?query_by=name"
        f"&sort_by=_text_match:desc,_vector_query(Embedding:([{vector_str}]),distance_threshold:0.3):asc"
        f"&exclude_fields=Embedding"
        f"&q=*"
    )

    try:
        response = requests.get(url_0, headers=HEADERS)
        response.raise_for_status()
        print("Success:", response.json())
    except requests.exceptions.RequestException as e:
        print("Error:", e)
        print("Response:", response.text)

    try:
        response = requests.get(url_1, headers=HEADERS)
        response.raise_for_status()
        print("Success:", response.json())
    except requests.exceptions.RequestException as e:
        print("Error:", e)
        print("Response:", response.text)

if __name__ == "__main__":
    test_vector_query()
```

## Expected Behavior

Ideally the argument should work just like when doing regular vector search or hybrid search using vector_query().

## Actual Behavior

The arguments doesn't get passed correctly. Suspicion is that the comma introduced by using the distance argument, triggers the parsing error.

Error: 400 Client Error: Bad Request for url: https://XXXXXXXXX/collections/XXXXXXXX/documents/search?query_by=name&sort_by=_text_match:desc,_vector_query(Embedding:(%5B0.0,0.0,0.0,0.0,0.0%5D),distance_threshold:0.3):asc,rank_score:desc&exclude_fields=Embedding&q=*
Response: {"message": "Only upto 3 sort fields are allowed."}

Error: 404 Client Error: Not Found for url: https://XXXXXXXXX/collections/XXXXXXXXX/documents/search?query_by=name&sort_by=_text_match:desc,_vector_query(Embedding:(%5B0.0,0.0,0.0,0.0,0.0%5D),distance_threshold:0.3):asc&exclude_fields=Embedding&q=*
Response: {"message": "Could not find a field named `_vector_query(Embedding` in the schema for sorting."}

## Metadata
typesense V26. 

typesense cloud

python 3.11

**Comment by PeterThramkrongart** [comment]
@kishorenc  Has the bug already been fixed, or has it just been added to the roadmap? 

The bug is also present in v27.rc35 btw...

**Comment by kishorenc** [comment]
Fixed in `27.0` GA release. Please reopen if you think there is still an issue.
